js/lightbeam.js

```diff
@@ -20,6 +24,72 @@ const lightbeam = {
     });
   },
 
+  // Called from init() (isFirstParty = undefined)
+  // and redraw() (isFirstParty = true or false).
+  async updateVars(isFirstParty) {
+    const numFirstParties = document.getElementById('num-first-parties');
+    const numThirdParties = document.getElementById('num-third-parties');
+
+    // initialize dynamic vars from storage
+    if (!this.dataGatheredSince) {
+      const { dateStr, fullDateTime } = await this.getDataGatheredSince();
+      this.dataGatheredSince = dateStr;
+      const dataGatheredSince = document.getElementById('data-gathered-since');
+      dataGatheredSince.textContent = this.dataGatheredSince;
+      dataGatheredSince.setAttribute('datetime', fullDateTime);
+    }
+    if (isFirstParty === undefined) {
+      this.numFirstParties = await this.getNumFirstParties();
+      numFirstParties.textContent
+        = (this.numFirstParties === 0) ? '' : `${this.numFirstParties} Sites`;
+      this.numThirdParties = await this.getNumThirdParties();
+      numThirdParties.textContent
+        = (this.numThirdParties === 0) ? ''
```

Comment:
Looks like these strings are used in multiple places too, please make it a function if this needs to be the case.